packages/appsync-modelgen-plugin/src/utils/process-index.ts

```diff
@@ -28,3 +30,14 @@ export const processIndex = (model: CodeGenModel) => {
   const deDupedKeyList = keyList.filter(key => !existingIndexNames.includes(key.arguments.name));
   model.directives.push(...deDupedKeyList);
 };
+
+/*
+ * Accepts a model and field name, and potentially empty list of sortKeyFields to generate a unique index name.
+ * e.g. modelName = Employee, fieldName = manager, sortKeyFields = [level]
+ * will generate a name like employeeByManagerAndLevel.
```

Comment:
nit: Wouldn't this be `employeesByManagerAndLevel` since you `pluralize` the model name?

Reply:
That's correct. Updated the name.